### ISSUE TITLE
Calls to redis with aioredis fail when profiling is enabled

### ISSUE THREAD

**Issue by mdespriee** [DataDog/dd-trace-py#3837]
### Which version of dd-trace-py are you using?

1.1.0

### Which version of the libraries are you using?

(Note: our package manager is conda.)
``` 
aioredis=2.0.1=pyhd8ed1ab_0
ddtrace=1.1.0=py39h5a03fae_1
python=3.9.12=h12debd9_1
... 
```

### How can we reproduce your problem?

no reproducer yet, will try to work on one.

### What is the result that you get?

When we enable the profiling (via env var `DD_PROFILING_ENABLED=true`), our calls to redis with `aioredis` client fails with this stack trace:

```
builtins.AttributeError: __aenter__
Traceback (most recent call last):
  File "/opt/conda/lib/python3.9/site-packages/ddtrace/contrib/redis/util.py", line 87, in _trace_redis_cmd
    yield span
  File "/opt/conda/lib/python3.9/site-packages/ddtrace/contrib/aioredis/patch.py", line 71, in traced_execute_command
    return await func(*args, **kwargs)
  File "/opt/conda/lib/python3.9/site-packages/aioredis/client.py", line 1082, in execute_command
    conn = self.connection or await pool.get_connection(command_name, **options)
  File "/opt/conda/lib/python3.9/site-packages/aioredis/connection.py", line 1407, in get_connection
    async with self._lock:
AttributeError: __aenter__
```

With profiling disabled, we don't have any issue.

**Comment by brettlangdon** [comment]
hey @mdespriee can you try upgrading to `ddtrace>=1.1.2`?

This should be addressed by #3685

https://ddtrace.readthedocs.io/en/stable/release_notes.html#v1-1-2

**Comment by mdespriee** [comment]
@brettlangdon indeed it's fixed in 1.1.2. Thanks
